app/gui/src/project-view/components/GraphEditor/widgets/WidgetSelection.vue

```diff
@@ -465,8 +482,11 @@ export interface Actions {
    *
    * For example, the {@link WidgetCloudBrowser} installs a custom entry that, when clicked,
    * opens a file browser where the dropdown was.
+   * @param keepAlive - when set, the `activity` instance will be kept between drop-down closing
+   *  and opening. The activity component must not change it type (when being a ref) and provide
+   * `name` option explicitly.
```

Comment:
An alternative approach would be `<KeepAlive v-if="..."><component ...></KeepAlive><component v-else ...>` (This could be packaged into a `ConditionalKeepAlive` slotted utility component to avoid writing the `<component ....>` twice). This would avoid the `name` requirement and the restriction on changing types.

Reply:
when `v-else` is hit, the cache will be gone I'm afraid...

...unless we put this in _another_ KeepAlive.

Reply:
I imagined `keepAlive` be requested _per activity_, so for example we have FileBrowser activity which is kept alive, but same drop-down could have another activity which is not: but even when that second activity is set, the FileBrowser is still cached.